lib/looker.ts

```diff
@@ -0,0 +1,59 @@
+import { LookerNodeSDK } from "@looker/sdk-node"
+import { IDashboardElement, IWriteQuery } from "@looker/sdk"
+
+/**
+ *
+ * @type {string} Local configuration file name, one directory above
+ */
+const SDK = LookerNodeSDK.init40()
+const me = await SDK.ok(SDK.me());
+
+export async function getAWDashboardElement0(): Promise<IDashboardElement> {
+  const dashboardId = "1471";
+//  await(getDBFilters())
+
+  const elements: IDashboardElement[] = await SDK.ok(SDK.dashboard_dashboard_elements(dashboardId));
+  return elements[0];
+}
+
+export async function runEventCountQuery(filters: any): Promise<any>{
+
+  const element0 = await getAWDashboardElement0()
+  const origQuery = element0.query as IWriteQuery
+
+  // take the query from the original dashboard
+  const newQueryBody = structuredClone(origQuery)
+  delete newQueryBody.client_id // must be unique per-query
+
+  // override the filters
+  newQueryBody.filters = Object.assign(
+    {
+      'event_counts.message_id': '%FAKESPOT_OPTIN_DEFAULT%',
+      'event_counts.submission_timestamp_date': '2 days'
+    },
+    filters
+  )
+
+  // console.log("filters: ", filters)
+  // console.log("newQueryBody.filters: ", newQueryBody.filters)
+
+
+  const newQuery = await SDK.ok(SDK.create_query(newQueryBody));
+  const result = await SDK.ok(SDK.run_query({
+      query_id: newQuery.id!,
+      result_format: "json"
+    }))
+
+  // console.log("newQueryBody.filters: ", newQueryBody.filters)
+  // console.log(" newQuery result: ", result)
+  return result
+}
+
+export async function setCTRPercent(id: string, template?: string): Promise<number|undefined> {
+  const queryResult = await runEventCountQuery(
+    { 'event_counts.message_id':  '%' + id + '%' }
+  )
+  if (queryResult.length > 0 && template !== 'infobar') {
+    return Number(Number(queryResult[0].primary_rate * 100).toFixed(1))
```

Comment:
I think the only reason it fails in infobar is because I didn't call it primary_rate there.  The API Explorer could help you figure this out. The best fix would be to fork the infobar dashboard and change the name in the new copy.

Reply:
I also noticed that the data from today is included in the infobar dashboard, which shows 0 impressions which gives a 0% CTR. So currently investigating that 👀

Reply:
If you haven't already done it, the InfoBar fix could be a separate bug/ticket so that it doesn't slow down this one.

Reply:
I wonder if the infobar dashboard problem is another case-sensitivity thing

Reply:
Yeah, I think I'll file a separate bug for this. I was able to make the dashboard show the last 30 complete days to remove today's incomplete data, but it was still giving a 0% CTR and I wasn't able to figure out why.